From: Jiang Xin <jiangxin@ossxp.com>
Subject: [PATCH] t/mirror_switch

Do not mirror repository if set gitolite.mirror to false.

Signed-off-by: Jiang Xin <jiangxin@ossxp.com>

---
 hooks/common/post-receive.mirrorpush |    7 +++++++
 1 个文件被修改，插入 7 行(+)

diff --git a/hooks/common/post-receive.mirrorpush b/hooks/common/post-receive.mirrorpush
index 4b31741..cba8b8a 100755
--- a/hooks/common/post-receive.mirrorpush
+++ b/hooks/common/post-receive.mirrorpush
@@ -20,4 +20,11 @@ die() { echo $0 "(post-receive.mirrorpush):" "$@" >&2; exit 1; }
 
 slaves=`git config --get gitolite.mirror.slaves`
 [ -z "$slaves" ] && exit 0
+if [ "$(git config --bool gitolite.mirror.disabled)" = "true" ]
+then
+    echo "**************************************************"
+    echo "WARNING: mirror disabled for this repository."
+    echo "**************************************************"
+    exit 0
+fi >&2
 $GL_BINDIR/gl-mirror-push $GL_REPO $slaves
-- 
tg: (f73e3d7..) t/mirror_switch (depends on: upstream)
